source/fe/mapping_q_generic.cc

```diff
@@ -3810,19 +3802,53 @@ add_quad_support_points(const Triangulation<3,3>::cell_iterator &cell,
                ExcInternalError());
 #endif
 
-      // ask the boundary/manifold object to return intermediate points on it
-      get_intermediate_points_on_object(face->get_manifold(), line_support_points,
-                                        face, quad_points);
-
-      // in 3D, the orientation, flip and rotation of the face might not
-      // match what we expect here, namely the standard orientation. thus
-      // reorder points accordingly. since a Mapping uses the same shape
-      // function as an FE_Q, we can ask a FE_Q to do the reordering for us.
-      for (unsigned int i=0; i<quad_points.size(); ++i)
-        a.push_back(quad_points[fe_q->adjust_quad_dof_index_for_face_orientation(i,
-                                face_orientation,
-                                face_flip,
-                                face_rotation)]);
+      // On a quad, we have to check whether the manifold should determine the
+      // point distribution or rather a weighted sum should be created. This
+      // is the same logic as in the compute_mapping_support_points function
+      // below
+      bool all_manifold_ids_are_equal = true;
+      for (unsigned int l=0; l<GeometryInfo<3>::lines_per_face; ++l)
+        if (&face->line(l)->get_manifold() != &face->get_manifold())
+          all_manifold_ids_are_equal = false;
```

Comment:
Oh, I misread -- you are of course only looking at those. Either way, your helper function should probably work here.